monitor/src/main/java/edu/asu/diging/monitor/web/admin/AddAppController.java

```diff
@@ -41,12 +42,12 @@ public String show(Model model) {
         }).collect(Collectors.toList()));
         return "admin/apps/show";
     }
-	
-	@RequestMapping(value="/admin/apps/add", method=RequestMethod.POST)
-	public String add(@ModelAttribute AppForm appForm) {
-		IApp app = new App();
-		appHelper.copyAppInfo(app, appForm);
-		appManager.addApp(app);
-		return "redirect:/admin/apps/add";
-	}
+
+    @RequestMapping(value = "/admin/apps/add", method = RequestMethod.POST)
+    public String add(@ModelAttribute AppForm appForm, RedirectAttributes redirectAttrs) {
+        IApp app = new App();
+        appHelper.copyAppInfo(app, appForm);
+        appManager.addApp(app, appForm);
```

Comment:
all values should be brought over from the form into the app project at the same time (in the same class). In other words your controllers should not copy half of them and then the manager does the rest.

Reply:
Resolved this in MON-13